src/ape/api/networks.py

```diff
@@ -606,6 +607,18 @@ def config(self) -> PluginConfig:
     def _network_config(self) -> Dict:
         return self.config.get(self.name, {})
 
+    @cached_property
+    def gas_limit(self) -> Union[Literal["auto"], Literal["max"], int]:
+        value = self._network_config.get("gas_limit", "auto")
```

Comment:
I think all the validation that happens here is unneeded since it happens at the config parsing level now, so could we safely return from here now, like: 

```suggestion
        return self._network_config.get("gas_limit", "auto")
```

Reply:
Updated in 324d04c5 and also fixed the type hinting - apparently `Literal` supports lists so these can just read `Literal["auto", "max"]` instead of using two `Literal`s